### PR TITLE
Add clarifying note to ibft spec

### PR DESCRIPTION
**Pull request by piersy**
The note explains why the protocol needs round change certificates and prepared certificates. Something that is not immediately obvious when reading the spec.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit ed028ddfb41f1bfaa5254ec58ac3c4d2a2791ba7
<details><summary>coverage:  45.4% of statements across all listed packages</summary><br>
<pre>
coverage:  51.2% of statements in consensus/istanbul
coverage:  37.5% of statements in consensus/istanbul/announce
coverage:  54.5% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  52.4% of statements in consensus/istanbul/core
coverage:  45.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  64.4% of statements in consensus/istanbul/uptime
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 7d21d07140
</details>

File: consensus/istanbul/celo-ibft-spec.md [hunk 1]
```diff
@@ -448,3 +448,29 @@ function call to occur after the given duration.
...
+
+## Appendix 2: Notes
+
+### What do the prepared certificate and round change certificate do?
+
+They enforce that if it’s possible that a node could have confirmed a block in
+a round, then all remaining nodes in future rounds can only confirm the same
+block.
+
+- A node could have confirmed a block if 2f+1 nodes send commit messages for a
```
Comment (palango) [inline consensus/istanbul/celo-ibft-spec.md:460]: sen**d** or sen**t**?

**Review by palango — APPROVED** [review]
Great, thanks for adding this!

One grammar nitpick, but you're the native speaker ;)

File: consensus/istanbul/celo-ibft-spec.md (hunk 1 above)
Comment (piersy) [reply consensus/istanbul/celo-ibft-spec.md:460]: Yes, I think sent is actually correct, thanks! :)

**Pull request merged**